disperser/apiserver/server.go

```diff
@@ -443,24 +443,25 @@ func (s *DispersalServer) GetBlobStatus(ctx context.Context, req *pb.BlobStatusR
 
 	requestID := req.GetRequestId()
 	if len(requestID) == 0 {
-		return nil, errors.New("invalid request: request_id must not be empty")
+		return nil, common.NewInvalidArgError("request_id must not be empty")
 	}
 
 	s.logger.Info("received a new blob status request", "requestID", string(requestID))
 	metadataKey, err := disperser.ParseBlobKey(string(requestID))
 	if err != nil {
-		return nil, err
+		return nil, common.NewInvalidArgError(fmt.Sprintf("failed to parse the requestID: %s", err.Error()))
 	}
 
 	s.logger.Debug("metadataKey", "metadataKey", metadataKey.String())
 	metadata, err := s.blobStore.GetBlobMetadata(ctx, metadataKey)
 	if err != nil {
-		return nil, err
+		// TODO: we need to distinguish NOT_FOUND from actual internal error.
```

Comment:
Would probably need to standarize the error message and then check that here.

Reply:
Yep, we could check the message, or surface an error with `NewNotFoundError` inside the GetBlobMetaata (this approach was used in the Churner, L95 in churner/server.go)